presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -317,7 +317,7 @@ private void processNewSources()
                 }
 
                 // if current operator is finished...
-                if (current.isFinished()) {
+                if (current.isFinished() && !next.isFinished() && isBlocked(next).isDone()) {
```

Comment:
I can drop `&& !next.isFinished()` from PR, if you don't like this part of the change.